kotlinx-coroutines-core/common/src/flow/terminal/Logic.kt

```diff
@@ -0,0 +1,107 @@
+@file:JvmMultifileClass
+@file:JvmName("FlowKt")
+
+package kotlinx.coroutines.flow
+
+import kotlinx.coroutines.*
+import kotlin.jvm.*
+
+
+/**
+ * A terminal operator that returns `true` and immediately cancels the flow
+ * if at least one element matches the given [predicate].
+ *
+ * If the flow does not emit any elements or no element matches the predicate, the function returns `false`.
+ *
+ * Equivalent to `!all { !predicate(it) }` (see [Flow.all]) and `!none { predicate(it) }` (see [Flow.none]).
+ *
+ * Example:
+ *
+ * ```
+ * val myFlow = flow {
+ *   repeat(10) {
+ *     emit(it)
+ *   }
+ *   throw RuntimeException("You still didn't find the required number? I gave you ten!")
+ * }
+ * println(myFlow.any { it > 5 }) // true
+ * println(flowOf(1, 2, 3).any { it > 5 }) // false
+ * ```
+ *
+ * @see Iterable.any
+ * @see Sequence.any
+ */
+public suspend fun <T> Flow<T>.any(predicate: suspend (T) -> Boolean): Boolean {
+    var found = false
+    collectWhile {
+        predicate(it).also { satisfies ->
+            if (satisfies) found = true
+        }.let(Boolean::not)
```

Comment:
Can we do something that is easier to read like:

```suggestion
        val satisfies = predicate(it)
        if (satisfies) found = true
        !satisfies
```